source/docs/software/wpilib-tools/path-planning/wpilib-trajectories/ramsete.rst

```diff
@@ -0,0 +1,55 @@
+Ramsete Controller
+==================
+The Ramsete Controller is a trajectory tracker that is built in to WPILib. This tracker can be used to accurately track trajectories with correction for minor disturbances.
+
+Constructing the Ramsete Controller object
+------------------------------------------
+The Ramsete controller should be initialized with two gains, namely ``b`` and ``zeta``. Larger values of ``b`` make convergence more aggressive like a proportional term whereas larger values of ``zeta`` provide more damping in the response. These controller gains only dictate how the controller will output adjusted velocities. It does NOT affect the actual velocity tracking of the robot. This means that these controller gains are generally robot-agnostic.
+
+.. note:: Gains of ``2.0`` and ``0.7`` for ``b`` and ``zeta`` have been tested repeatedly to produce desirable results.
+
+Getting adjusted velocities
+---------------------------
+The Ramsete controller returns "adjusted velocities" so that the when the robot tracks these velocities, it accurately reaches the goal point. The controller should be updated periodically with the new goal. The goal comprises of a desired pose, desired linear velocity, and desired angular velocity. Furthermore, the current position of the robot should also be updated periodically. The controller uses these four arguments to return the adjusted linear and angular velocity. Users should command their robot to these linear and angular velocities to achieve optimal trajectory tracking.
+
+.. note:: The "goal pose" represents the position that the robot should be at at a particular timestep when tracking the trajectory. It does NOT represent the final endpoint of the trajectory.
+
+The controller can be updated using the ``Calculate`` (C++) / ``calculate`` (Java) method. There are two overloads for this method. Both of these overloads accept the current robot position as the first parameter. For the other parameters, one of these overloads takes in the goal as three separate parameters (pose, linear velocity, and angular velocity) whereas the other overload accepts a ``Trajectory.State`` object, which contains information about the goal pose. For its ease, users should use the latter method when tracking trajectories.
+
+.. tabs::
+
+   .. code-tab:: java
+
+      var goal = trajectory.sample(3.4); // sample the trajectory at 3.4 seconds from the beginning
```

Comment:
I'd prefer you to not use var in partial examples. We want to see what exactly this object is, and FRC Java teams are often not familiar with ``var``

Reply:
Where is this being run, in teleopPeriodic?

Reply:
Added a comment clarifying that it should be run periodically. It can either be in the `execute` method of a command, the `periodic` method of a subsystem or in the `teleopPeriodic` method -- that implementation is up to the user.

Reply:
Fixed the `var` usage.